sync-with-upstream/entrypoint.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -eux
+
+#
+# This variable must be passed into this script.
+#
+UPSTREAM_REPOSITORY="$1"
+UPSTREAM_BRANCH="$2"
+DOWNSTREAM_BRANCH="$3"
+
+[[ -n "${UPSTREAM_REPOSITORY}" ]] || exit 1
+[[ -n "${UPSTREAM_BRANCH}" ]] || exit 1
+[[ -n "${DOWNSTREAM_BRANCH}" ]] || exit 1
+
+#
+# We need these config parameters set in order to do the git-merge.
+#
+git config user.name "${GITHUB_ACTOR}"
+git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+#
+# We need the full git repository history in order to do the git-merge.
+#
+git fetch --unshallow
+
+git remote add upstream "$UPSTREAM_REPOSITORY"
+git fetch upstream
+
+if git show-ref --verify --quiet "refs/heads/${DOWNSTREAM_BRANCH}"; then
+	git checkout "${DOWNSTREAM_BRANCH}"
+else
+	git checkout -b "${DOWNSTREAM_BRANCH}"
+fi
```

Comment:
It's been a little while, so it's hard to say for sure (I can't fully remember).. but I think that's correct. It's so we'll automatically get a message like:
```
merge $UPSTREAM_BRANCH into $DOWNSTREAM_BRANCH
```
rather than using arbitrary local branch names in the commit message, I think it's cleaner and more informative to refer directly to the remote branch names being used in the merge.

And while we could change the commit message to anything we'd like, after the fact, I prefer to use the default automatic message generated by `git merge` (and configuring the local repository in order to manipulate that automatic message).

Reply:
Yeah that makes sense, just wanted to confirm that this is what went into the reasoning.